src/operator/account/ConvertAccountToContractOverlay.tsx

```diff
@@ -0,0 +1,266 @@
+import moment from 'moment'
+
+const noOp = () => {}
+
+const ConvertAccountToContractOverlay: FC = () => {
+  const {
+    account,
+    handleConvertAccountToContract,
+    organizations,
+    setConvertToContractOverlayVisible,
+    convertToContractOverlayVisible,
+  } = useContext(AccountContext)
+
+  const convertAccountToContract = () => {
+    if (startDateInputStatus !== ComponentStatus.Error) {
+      try {
+        handleConvertAccountToContract(startDateInput)
+        setConvertToContractOverlayVisible(false)
+      } catch (e) {
+        setConvertToContractOverlayVisible(false)
+      }
+    }
+  }
+
+  const dateFormat = 'YYYY-MM-DD'
+  const convertToMomentDate = (date: Date | string) => moment(date, true)
+  const isValidDate = (date: Date | string) =>
+    convertToMomentDate(date).isValid()
+  const formatDate = (date: Date | string) =>
+    convertToMomentDate(date).format(dateFormat)
+
+  const [startDateInput, setStartDateInput] = useState<string>(
+    formatDate(new Date())
+  )
+
+  const [startDateInputStatus, setStartDateInputStatus] = useState<
+    ComponentStatus
+  >(ComponentStatus.Default)
+
+  const [isDatePickerOpen, setIsDatePickerOpen] = useState<boolean>(false)
+
+  const [
+    isOnClickOutsideHandlerActive,
+    setIsOnClickOutsideHandlerActive,
+  ] = useState<boolean>(true)
+
+  const triggerRef: RefObject<ButtonRef> = createRef()
+
+  const handleInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setStartDateInput(event.target.value)
+  }
+
+  const handleBlur = () => {
+    if (isValidDate(startDateInput)) {
+      setStartDateInputStatus(ComponentStatus.Default)
+    } else {
+      setStartDateInputStatus(ComponentStatus.Error)
+    }
+  }
+
+  const getDatePickerDateTime = () => {
+    const date = convertToMomentDate(startDateInput)
+    return !Number.isNaN(date.valueOf())
+      ? date.toISOString()
+      : new Date().toISOString()
+  }
+
+  const handleReset = () => {
+    setStartDateInput('')
+    setStartDateInputStatus(ComponentStatus.Default)
+  }
+
+  const showDatePicker = () => setIsDatePickerOpen(true)
+  const hideDatePicker = () => setIsDatePickerOpen(false)
+
+  const toggleDatePicker = () => {
+    if (isDatePickerOpen) {
+      hideDatePicker()
+    } else {
+      showDatePicker()
+    }
+  }
+
+  const handleSelectDate = (date: string) => {
+    if (isValidDate(date)) {
+      const formattedDate = formatDate(date)
+      setStartDateInput(formattedDate)
+      setStartDateInputStatus(ComponentStatus.Default)
+    } else {
+      setStartDateInput('')
+      setStartDateInputStatus(ComponentStatus.Error)
+    }
+  }
+
+  const onClickOutside = () => {
+    if (isOnClickOutsideHandlerActive) {
+      hideDatePicker()
+    }
+  }
+
+  const allowOnClickOutside = () => setIsOnClickOutsideHandlerActive(true)
+  const suppressOnClickOutside = () => setIsOnClickOutsideHandlerActive(false)
+
+  const styles: CSSProperties = isDatePickerOpen
```

Comment:
Not that much css here, but I wonder whether this is more readable if we do something like `className={x ? x : y}`, and make the styles live in a separate .scss file.

Reply:
I'm going to open an issue/feature request in [clockface](https://github.com/influxdata/clockface) around the datepicker, which will suggest a number of improvements to the built-in datepicker that I'm hoping will allow most of this code to go away. The main reason I kept a number of these things the way they are is to keep it more consistent with `TimeTickInput`, which will allow for easier refactoring with a new and improved `DatePicker` component that ideally eliminates the need for so much of this extra code, like error handling, manually toggling the datepicker overlay, and these styles.

But realistically, even if I end up spending some time working on clockface's datepicker to get a PR open, it likely won't be done for a little while at best, so I'm okay with whatever aligns this code with our ui development patterns, so if that means adding a separate styles file, we can do that. I definitely prefer that approach personally.

Reply:
It's not a lot of css, so I think it looks good as is without moving into a scss file. Keeping this as is for consistency with `TimeTickInput` makes a lot of sense to me.